tool/deploy_webapp.sh

```diff
@@ -5,37 +5,70 @@ if [ $# -ne 2 ]; then
     exit 1
 fi
 
-########## cd to the nook project directory and get the absolute path
-
-cd "$(dirname "$0")"/..
-PROJDIR="$(pwd)"
+WORK_DIR="$(pwd)"
 
 ########## validate argument(s)
 
 FIREBASE_CONSTANTS="$1"
 if [ ! -f "$FIREBASE_CONSTANTS" ]; then
   echo "could not find FIREBASE_CONSTANTS: $FIREBASE_CONSTANTS"
-  echo "  in: $(pwd)"
   exit 1
 fi
 
-CRYPTO_TOKEN_FILE="$2"
-if [ ! -f "$CRYPTO_TOKEN_FILE" ]; then
-  echo "could not find CRYPTO_TOKEN_FILE: $CRYPTO_TOKEN_FILE"
-  echo "  in: $(pwd)"
+cd "$(dirname "$FIREBASE_CONSTANTS")"
+FIREBASE_CONSTANTS_DIR="$(pwd)"
+FIREBASE_CONSTANTS_FILENAME="$(basename "$FIREBASE_CONSTANTS")"
+FIREBASE_CONSTANTS_FILE="$FIREBASE_CONSTANTS_DIR/$FIREBASE_CONSTANTS_FILENAME"
+
+
+CRYPTO_TOKEN="$2"
+if [ ! -f "$CRYPTO_TOKEN" ]; then
+  echo "could not find CRYPTO_TOKEN: $CRYPTO_TOKEN"
+  exit 1
+fi
+
+cd "$(dirname "$CRYPTO_TOKEN")"
+CRYPTO_TOKEN_DIR="$(pwd)"
+CRYPTO_TOKEN_FILENAME="$(basename "$CRYPTO_TOKEN")"
+CRYPTO_TOKEN_FILE="$CRYPTO_TOKEN_DIR/$CRYPTO_TOKEN_FILENAME"
+
+########## validate that the firebase constants file and the crypto token point to the same project
+
+# Get the project id from the firebase constants file
+echo "getting project id from firebase constants..."
+FIREBASE_CONSTANTS_PROJECT_ID=$(cat $FIREBASE_CONSTANTS_FILE | python -c 'import json,sys; constants=json.load(sys.stdin); print(constants["projectId"])')
+echo "project id: $FIREBASE_CONSTANTS_PROJECT_ID"
+
+# Get the project id from the crypto token file
+echo "getting project id from crypto token..."
+CRYPTO_TOKEN_PROJECT_ID=$(cat $CRYPTO_TOKEN_FILE | python -c 'import json,sys; constants=json.load(sys.stdin); print(constants["project_id"])')
+echo "project id: $CRYPTO_TOKEN_PROJECT_ID"
+
+if [ "$FIREBASE_CONSTANTS_PROJECT_ID" != "$CRYPTO_TOKEN_PROJECT_ID" ]; then
+  echo "the two project ids must match, check that you're deploying to the correct project"
   exit 1
 fi
 
+########## cd to the nook project directory and get the absolute path
+
+cd "$WORK_DIR"
+
+cd "$(dirname "$0")"/..
+PROJECT_DIR="$(pwd)"
```

Comment:
Should we rename this "NOOK_DIR" to keep the same env var naming conventions as in the KK project scripts?

Reply:
Good idea, done!